### PR TITLE
Hans/upgrade insights client to 0.10 for ec3 reportsconfig lca

### PR DESCRIPTION
**Pull request by hansrajkvb**
Upgrade insights client from version 0.4.0 to 0.10.1 for EC3, reports-config and one click LCA widgets.
Fixed function calls and tests to reflect new function signatures.

### CODE REVIEW

**Comment by Guillar1** [comment]
We've discussed about the format issues.

File: packages/itwin/ec3-widget/src/components/Templates.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ export const Templates = ({ onClickCreate, onClickTemplateTitle }: TemplateProps
           displayName: x.displayName,
           description: x.description ?? "",
           id: x.id,
-          labels: x.labels,
+          labels: [],
```
Comment (Guillar1) [inline packages/itwin/ec3-widget/src/components/Templates.tsx:57]: Is this intentional?

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts [hunk 2]
```diff
@@ -150,39 +145,45 @@ export class BulkExtractor {
 
   public async runReportExtractions(reportIds: string[]): Promise<void> {
     const reportIModelIds = new Map<string, string[]>();
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts:147]: Can you remove this `reportIModelIds`. Looks redundant.

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts [hunk 3]
```diff
@@ -150,39 +145,45 @@ export class BulkExtractor {
...
+    const extractionDetails: ExtractionRequestDetails[] = Array.from(
+      extractionDetailsIModelIdMap.entries()).map(([iModelId, mappings]) => {
+      const mappingSetArray: ExtractionMapping[] = Array.from(new Set(mappings.map((m) => m.id))).map((id) => ({id}));
+      return {
+        iModelId,
+        mappings: mappingSetArray,
+      };
+    });
+    for (const extractionDetail of extractionDetails) {
+      await this.runIModelExtractions([extractionDetail]);
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts:170]: Is this for loop necessary?

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts [hunk 4]
```diff
@@ -150,39 +145,45 @@ export class BulkExtractor {
...
+          extractionDetailsIModelIdMap.get(extractionDetail.iModelId)?.concat(extractionDetail.mappings) || extractionDetail.mappings);
+      });
     }
-    const iModels = new Set(Array.from(reportIModelIds.values()).flat());
 
-    for (const iModel of iModels) {
-      await this.runIModelExtractions([iModel]);
+    const extractionDetails: ExtractionRequestDetails[] = Array.from(
+      extractionDetailsIModelIdMap.entries()).map(([iModelId, mappings]) => {
+      const mappingSetArray: ExtractionMapping[] = Array.from(new Set(mappings.map((m) => m.id))).map((id) => ({id}));
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts:163]:
```
const existingMappings = extractionDetailsIModelIdMap.get(detail.iModelId) || [];
extractionDetailsIModelIdMap.set(detail.iModelId, [...existingMappings, ...detail.mappings]);
```

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts [hunk 5]
```diff
@@ -150,39 +145,45 @@ export class BulkExtractor {
...
-      const reportIModels = await this.fetchReportIModels(reportId);
+      const reportExtractionDetails = await this.fetchReportExtractionRequestDetails(reportId);
+      const reportIModels = reportExtractionDetails.map((reportExtractionDetail) => reportExtractionDetail.iModelId);
       reportIModelIds.set(reportId, reportIModels);
       this._reportIModels.set(reportId, reportIModels);
+      reportExtractionDetails.forEach((extractionDetail) => {
+        extractionDetailsIModelIdMap.set(
+          extractionDetail.iModelId,
+          extractionDetailsIModelIdMap.get(extractionDetail.iModelId)?.concat(extractionDetail.mappings) || extractionDetail.mappings);
+      });
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts:158]:
```
const existingMappings = extractionDetailsIModelIdMap.get(extractionDetail.iModelId) || [];
extractionDetailsIModelIdMap.set(extractionDetail.iModelId, [...existingMappings, ...extractionDetail.mappings]);
```

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts [hunk 6]
```diff
@@ -191,22 +192,28 @@ export class BulkExtractor {
...
+    for (const extractionRequestDetails of extractionRequestsDetails) {
+      const run = await this.runExtraction(extractionRequestDetails);
       if (run) {
-        this._iModelStates.set(iModelId, ExtractionState.Queued);
-        this._iModelRun.set(iModelId, run);
+        this._iModelStates.set(extractionRequestDetails.iModelId, ExtractionState.Queued);
+        this._iModelRun.set(extractionRequestDetails.iModelId, run);
       }
       this.checkRunning();
     }
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts:203]: You can probably wrap this in a `Promise.all`. 
```
    await Promise.all(extractionRequestsDetails.map(async details => {
      const run = await this.runExtraction(details);
      if (run) {
        this._iModelStates.set(details.iModelId, ExtractionState.Queued);
        this._iModelRun.set(details.iModelId, run);
      }
    }));
```

File: packages/itwin/reports-config-widget/src/test/ReportMappings.test.tsx [hunk 7]
```diff
@@ -20,12 +21,12 @@ import {
 import * as moq from "typemoq";
 import type {
   MappingContainer,
+  MappingsClient,
   Report,
-  ReportMappingCollection,
-} from "@itwin/insights-client";
+
```
Comment (Guillar1) [inline packages/itwin/reports-config-widget/src/test/ReportMappings.test.tsx:26]: Empty line.

**Review by saskliutas — APPROVED** [review]
Approving `test-viewer` part

File: packages/itwin/ec3-widget/src/components/Templates.tsx (hunk 1 above)
Comment (Guillar1) [reply packages/itwin/ec3-widget/src/components/Templates.tsx:57]: Seems like labels isn't even being used in this component. Eventually we are going to have to switch to the provided types in `insights-client`. That's for another day.

File: packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts (hunk 3 above)
Comment (hansrajkvb) [reply packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts]: Fixed

**Review by Guillar1 — APPROVED** [review]

**Pull request merged**